### PR TITLE
[Rename] Remove "diem" pattern from various directories.

### PR DESCRIPTION
**Pull request by JoshLind**
Note: most noticeably, this PR will break metric dashboards that use the old metric names.

## Motivation

This PR goes through most of the directories in the `aptos-core` codebase and replaces "diem" with "aptos" where we are able. Several directories have not been touched because of the nature of the changes (e.g., heavy involvement with the Move framework): `aptos-move`, `crates`, `scripts` and `sdk`. These can be updated in a follow-up PR.

For all other directories, name replacement occurs where appropriate/available. However, if there's anything Move or Framework related, this is not modified. So, we'll likely need a final sweep of everything once all Move/framework changes are made.

The PR offers a commit for each directory. Some changes bleed across directories (e.g., a variable is defined in one and used in another -- thanks to code refactoring in IDEs!).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Manual inspection.

## Related PRs

None.

### CODE REVIEW

**Review by davidiw — APPROVED** [review]
nice

**Review by zekun000 — APPROVED** [review]

File: types/src/chain_id.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ pub enum NamedChain {
 
 impl NamedChain {
     fn str_to_chain_id(s: &str) -> Result<ChainId> {
-        // TODO implement custom macro that derives FromStr impl for enum (similar to diem/common/num-variants)
+        // TODO implement custom macro that derives FromStr impl for enum (similar to aptos/common/num-variants)
```
Comment (gregnazario) [inline types/src/chain_id.rs:28]: aptos-core

**Review by gregnazario — APPROVED** [review]
Just fix up the couple typos I found.  Let's see if anything breaks /canary

**Comment by JoshLind** [comment]
Going to land this. @FYI: @msmouse -- this is probably going to break the dashboards. I'll try and go through all the dashboards before the next deployment and fix things 😄 There's also a bunch of other dashboard fixes that need to happen...

**Pull request closed without merging**